src/hypothesis/extra/datetime.py

```diff
@@ -70,6 +70,40 @@ def do_draw(self, data):
                 pass
 
 
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_days=None, max_days=None):
```

Comment:
I see some problems with allowing to configure more than min/max days.
tl;dr I see possible inconsistent api (thanks to timedeltas api) and inconsistent requirements for the end user to do the math in one places, and not to do in others.

First of all, timedeltas do heavy normalization and provide only three attributes to work with: `days`, `seconds` and `microseconds`. If we allow to provide more than that, we will have no simple way to test if it's working properly - we will have to deal with this normalization to three attributes. I'd rather ask the end user to do the math if they want to.
Secondly, if the end user will want to filter out some values, they won't be able to do something like `timedeltas(hours=(9, 18)).filter(lambda x: is_traffic_spike(x.hours))` because there are no `hours` attribute - they will need to do the math anyway.